File: python/Problem014.py
#Start: 837799 Length: 524 in 32.21921443939209
import time

start = time.time()

maximum = 1000000
#start, sequence length
largest_pair = [0,0]

def collatz(start, length=0):
    while start != 1:
        if start % 2 == 0:
            start /= 2
        else:
            start = 3*start + 1
        length += 1
    return length

for i in range(1,maximum):
    c = collatz(i)
    if c > largest_pair[1]:
        largest_pair[0] = i
        largest_pair[1] = c

print("Start:",largest_pair[0],"Length:",largest_pair[1],"in",time.time()-start)

File: java/Problem014.java
package main;

public class Problem014 {
	public static void main(String args[]) {
		System.out.println(new Problem014().run());
	}

	public long run() {
		long longestChain = 0;
		int size = 0;
		for (long i = 1; i < 1000000; i++) {
			int c = collatz(i);
			if (c > size) {
				longestChain = i;
				size = c;
			}
		}
		return longestChain;
	}

	public int collatz(long n) {
		int length = 1;
		long start = n;
		while (n != 1) {
			if (n % 2 == 0) {
				n /= 2;
			} else {
				n = n * 3 + 1;
			}
			length++;
		}
		return length;
	}
}
